src/modules/samples/cpp/src/so/SampleModule.cpp

```diff
@@ -0,0 +1,395 @@
+                status = ENOMEM;
+            }
+            else
+            {
+                std::fill(*payload, *payload + buffer.GetSize(), 0);
+                std::memcpy(*payload, buffer.GetString(), buffer.GetSize());
+                *payloadSizeBytes = buffer.GetSize();
+            }
+        }
+        catch (const std::exception& e)
+        {
+            OsConfigLogError(SampleLog::Get(), "Could not allocate payload: %s", e.what());
+            status = EINTR;
+
+            if (nullptr != *payload)
+            {
+                delete[] * payload;
+                *payload = nullptr;
+            }
+
+            if (nullptr != payloadSizeBytes)
+            {
+                *payloadSizeBytes = 0;
+            }
+        }
+    }
+
+    return status;
+}
+
+int MmiGet(
+    MMI_HANDLE clientSession,
+    const char* componentName,
+    const char* objectName,
+    MMI_JSON_STRING* payload,
+    int* payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    ScopeGuard sg{[&]()
+    {
+        if (IsFullLoggingEnabled())
+        {
+            if (MMI_OK == status)
+            {
+                OsConfigLogInfo(SampleLog::Get(), "MmiGet(%p, %s, %s, %.*s, %d) returned %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+            }
+            else
+            {
+                OsConfigLogError(SampleLog::Get(), "MmiGet(%p, %s, %s, %.*s, %d) returned %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+            }
+        }
+    }};
+
+    if (nullptr == clientSession)
+    {
+        OsConfigLogError(SampleLog::Get(), "MmiGet called with null clientSession");
+        status = EINVAL;
+        return status;
+    }
+
+    if (nullptr == payloadSizeBytes)
+    {
+        OsConfigLogError(SampleLog::Get(), "MmiGet called with null payloadSizeBytes");
+        status = EINVAL;
+        return status;
+    }
+
+    *payload = nullptr;
+    *payloadSizeBytes = 0;
+
+    Sample* session = reinterpret_cast<Sample*>(clientSession);
+
+    unsigned int maxPayloadSizeBytes = session->GetMaxPayloadSizeBytes();
+    rapidjson::Document document;
+
+    // Dispatch the get request to the appropriate handler for the given component and object
+    if (0 == g_componentName.compare(componentName))
+    {
+        if (0 == g_objectName.compare(objectName))
+        {
+            std::string value = session->GetValue();
+            document.SetString(value.c_str(), document.GetAllocator());
+
+            // Serialize the JSON object to the payload buffer
+            status = SerializeJsonPayload(document, payload, payloadSizeBytes, maxPayloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(SampleLog::Get(), "MmiGet called with invalid objectName: %s", objectName);
+            status = EINVAL;
+        }
+    }
```

Comment:
All this code needs to be in the static lib for maximum unittest coverage.

The dynamic object MmiGetInfo, MmiGet and MmiSet only FW the calls to the base class GetInfo, Get and Set methods